### PR TITLE
feat: allow asynchronous extractors

### PR DESCRIPTION
**Pull request by marstamm**
related to https://github.com/bpmn-io/internal-docs/issues/680

### CODE REVIEW

File: src/camunda-platform/index.js [hunk 1]
```diff
@@ -31,21 +31,26 @@ import { selfAndAllFlowElements, getParents, getElement } from '../shared/util/E
  * @returns {Array<ProcessVariable>}
```
Comment (philippfromme) [inline src/camunda-platform/index.js:31]: This function now returns a promise. We should adjust the JSDoc comments.

**Comment by marstamm** [comment]
We use this library in the properties panel and the modeler as of today. The use cases are slightly different:

- [C8 Properties panel](https://github.com/bpmn-io/bpmn-js-properties-panel/blob/dcbcdcd1f6e2daf152d8772cf1ec9b3d0c965a2c/src/provider/HOCs/withVariableContext.js): get list of variables to support in FEEL editing
- [C7 properties panel](https://github.com/bpmn-io/bpmn-properties-panel/blob/73701d26eb0b78ef9506bbf268f06cf76f2e37f5/src/provider/camunda-platform/properties/ProcessVariablesProps.js): List all available variables for an element, dependent on scope
- [Modeler](https://github.com/camunda/camunda-modeler/blob/master/client/src/util/metrics/processVariables.js): Metric that reports how many process variables are used

All places are either already async or can be adjusted to be async within the scope of their Preact component. We always use this library directly and not nested within another lib

**Comment by marstamm** [comment]
Correction on the [C7 properties panel](https://github.com/bpmn-io/bpmn-properties-panel/blob/73701d26eb0b78ef9506bbf268f06cf76f2e37f5/src/provider/camunda-platform/properties/ProcessVariablesProps.js) entry:

It is not used in a Component but in a synchronous `props` function, so integration there would not be straight forward. It currently needs to be synchronous so we can show/hide the group during rendering depending on available variables.

However, I believe with a small change in behavior this can be resolved.

- Always show `Process Variables` group if the element allows it ("no surprises")
- Introduce empty state if no variables are available

This way, variables can be fetched in a Component and don't need to be available before render.

**Comment by nikku** [comment]
Thanks for the details @marstamm. Good to go from my birds eye view them :rocket:

**Review by nikku — APPROVED** [review]

**Pull request merged**